Flames of Freedom package's recipe

wiz21b opened the pull request:
### Brief summary of what the package does

Displays the Flames of Freedom.
These are the eternal flames of freedom (and an homage to RMS who
is having tough times in this year 2019).
A small poem of yours can be displayed.

Resemble the fireplace package, but a bit more animated. 


### Direct link to the package repository

https://github.com/wiz21b/FlamesOfFreedom

### Your association with the package

Author and maintainer.

### Relevant communications with the upstream package maintainer

n/a

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [ I guess so] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

The package install that happens when I hit C-c C-c complains about missing version tags (although there are there and checkdoc/package-lint don't complain). The command "package-install-from-buffer" works fine.

alphapapa commented:
Cool!  If you have the time and interest, I'd welcome contributions to https://github.com/alphapapa/emacs-package-dev-handbook with specific tips to produce faster code.

wiz21b commented:
That'd be nice indeed, I may even have some time for that. OTOH I'm not sure I qualify for that job :-/ Indeed this package is the very first thing I've coded in elisp so I don't know if my experience is long enough for me to be able to give advice to others :-) But I'll look into it.

alphapapa commented:
It sounds like you did careful study and experimentation to find how to write certain code in Elisp to perform faster.  That's valuable research!  :)

wiz21b commented:
Dear alphapapa, I've submitted a PR in your https://github.com/alphapapa/emacs-package-dev-handbook . Feel free to comment.

riscy commented:
Very cool effect!

<kbd>M-x checkdoc</kbd> (using version 0.6.1):
No issues! 💯

<kbd>M-x package-lint-current-buffer</kbd> (using version 20191124.132):
```
3 issues found:

292:12: warning: Closing parens should not be wrapped onto new lines.
340:22: warning: Closing parens should not be wrapped onto new lines.
341:18: warning: Closing parens should not be wrapped onto new lines.
```

<kbd>M-x byte-compile-file</kbd> (using Emacs 26.3):
No byte-compilation issues! 😎

Other:
- flames-of-freedom.el#L6: Use `Version` instead of `Package-Version` (MELPA automatically adds `Package-Version`)
- flames-of-freedom.el#L268: No `format` required; messages are already f-strings
- flames-of-freedom.el#L334: No `format` required; messages are already f-strings
- reflexion -> reflection
- predictible -> predictable
- efficitenly -> efficiently
- coulours -> colours/colors

wiz21b commented:
I have fixed all the remarks you've noted. I have also fixed my release process a bit to avoid premature releases :-)

riscy commented:
It's preferable to continue to work on the default (`master`) branch if you can.  The defaults are always preferred! :)  (To avoid premature releases, I personally use a `develop` branch, but there is also the more official option to use [MELPA stable](https://github.com/melpa/melpa#melpa-stable) by creating versioned releases).

Also, my comment re: Package-Version basically means to change this:
```elisp
;; Package-Version: 1.1
```
Into this:
```
;; Version: 1.1
```
since MELPA will handle package-version.

purcell commented:
Yeah, I'd generally suggest working directly on `master`. Perhaps a topic branch here and there for ongoing experimental work. MELPA always wants to serve your latest `master` packages, and tagging will produce stable releases in MELPA Stable.

wiz21b commented:
I have reverted to master branch and fixed all the things you said (I missed some of those in the previous commit :-) ) Thanks for all the comments, it's very encouraging.

riscy commented:
Your recent commit added another recipe. :)

wiz21b commented:
Voilà, removed the unwanted additional recipe.

riscy commented:
Just in time for the holidays. :)  And, welcome to MELPA!
